service/internal/proctelemetry/config_test.go

```diff
@@ -123,6 +124,218 @@ func TestMetricReader(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "periodic/otlp-exporter-invalid-protocol",
+			reader: telemetry.MetricReader{
+				Periodic: &telemetry.PeriodicMetricReader{
+					Exporter: telemetry.MetricExporter{
+						Otlp: &telemetry.OtlpMetric{
+							Protocol: *strPtr("http/invalid"),
+						},
+					},
+				},
+			},
+			err: errors.New("unsupported protocol http/invalid"),
+		},
+		{
+			name: "periodic/otlp-grpc-exporter-no-endpoint",
+			reader: telemetry.MetricReader{
+				Periodic: &telemetry.PeriodicMetricReader{
+					Exporter: telemetry.MetricExporter{
+						Otlp: &telemetry.OtlpMetric{
+							Protocol:    "grpc/protobuf",
+							Compression: strPtr("gzip"),
+							Timeout:     intPtr(1000),
+							Headers: map[string]interface{}{
+								"test": "test1",
```

Comment:
Good point, i've opened a PR to the configuration repo to change this to a map of strings: https://github.com/open-telemetry/opentelemetry-configuration/pull/32

To make this a map of configopaque string i would need to run a substitution on the generated code.

Reply:
Well it turns out because of a limitation of go-jsonschema, we need a patch anyways. I've opened the PR to make this a map[string]string. I'll take a look at implementing this functionality in the library itself, but for now https://github.com/open-telemetry/opentelemetry-collector/pull/8113